kmk/scanners/digitalio.py

```diff
@@ -69,14 +71,15 @@ def __init__(
             pin.switch_to_output()
 
         for pin in self.inputs:
-            pin.switch_to_input(pull=digitalio.Pull.DOWN)
+            pin.switch_to_input(pull=self.pull)
 
         self.rollover_cols_every_rows = rollover_cols_every_rows
         if self.rollover_cols_every_rows is None:
             self.rollover_cols_every_rows = self.len_rows
 
         self._key_count = self.len_cols * self.len_rows
-        self.state = bytearray(self.key_count)
+        initial_state_value = b'\x01' if self.pull is digitalio.Pull.UP else b'\x00'
+        self.state = bytearray(initial_state_value) * self.key_count
```

Comment:
Is there a reason you're using a bytearray here? I tried something like `self.state = ([self.pull is digitalio.Pull.UP for i in range(self.key_count)])` in place of line 81 and 82 and it seemed to have the same behavior.

Reply:
I should add, I'm not saying one or the other is right. I'm new to this repo and actively testing this with an MCP23018 IO chip. I've been fiddling with things trying to understand the code.

Reply:
I honestly didn't put any thought into it. This pull request is focused on something else, so I decided to leave it at a bytearray for consistency.